### PR TITLE
fix turbomind stream canceling

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
May entitle this PR clearly

**Review by AllentDan — COMMENTED** [review]
The program crash still exists. Besides, is hard code input_ids to empty appropriate?

**Review by AllentDan — APPROVED** [review]
LGTM

**Comment by lvhan028** [comment]
May fix the conflicts

File: lmdeploy/turbomind/turbomind.py [hunk 1]
```diff
@@ -207,16 +205,20 @@ def _forward_callback(self, result, ctx):
         self.que.put((False, result))
 
     def _forward_thread(self, inputs):
+        instance_comm = self.tm_model.model_comm.create_instance_comm(
```
Comment (lvhan028) [inline lmdeploy/turbomind/turbomind.py:208]: If `create_instance_comm` being put to a local thread, will it cost more time?

Comment (grimoire) [reply lmdeploy/turbomind/turbomind.py:208]: https://github.com/InternLM/lmdeploy/blob/main/src/turbomind/models/llama/LlamaInstanceComm.h#L10
https://github.com/InternLM/lmdeploy/blob/main/src/turbomind/triton_backend/llama/LlamaTritonModel.cc#L388
InstanceComm only contain a pointer and a Barrier object (which is an adapter of pthread_barrier_t). The cost should be ignorable.

Comment (AllentDan) [reply lmdeploy/turbomind/turbomind.py:208]: > If `create_instance_comm` being put to a local thread, will it cost more time?

I tested it this morning. Did not influence performance.

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**